### PR TITLE
Nightmare Jaunt is delayed, but no longer revealed by light

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Nightmare's Jaunt has recieved two changes.

1. The Nightmare is no longer forced out of jaunt when their tile is lit up.
2. Jaunting now takes 2.5 seconds.
## Why It's Good For The Game

Nightmares, to be honest, are just not fun anymore, if they ever were. Their 'kit', that being 3 abilities, is not nearly powerful nor versatile enough to give them the slightest hint of a chance against either a coordinated security team or one robust validhunter. The vast majority of the time there's just no chance for them to do anything, because as soon as they're revealed 30 people are hunting them in maintenance with lanterns, or spamming untouchable glass floors, or glowshrooms. Their jaunt also has an incredibly maddening issue of randomly shunting you out in places where it probably shouldn't, which can end up trapping you outside the station or caught in a terrible spot to be unceremoniously hit with a baton twice and killed.

This change turns the Nightmare into a true ambush predator instead of some weird.. jaunt... guy. By giving them unlimited jaunting, they can pick and choose who to target and when, and can easily scour the station for any foolish assistants in maintenance while making sure not to be near any security. This makes them hard to catch, but during the time they're in jaunt they can't really do anything, so it's not a big deal.

The 2.5 second delay for jaunting is to prevent these guys from just turning into frickin' gods. They need to be aware of their position and can't use it as a get away from jail free card anymore, once more encouraging the ambush predator playstyle rather than the 'idk kill things until you die' style it vaguely had.
## Changelog
:cl:
balance: Nightmare Jaunt is delayed, but no longer revealed by light
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
ilight pr

**Comment by DrAmazing343** [comment]
I will say that this change, while good on paper, would turn any chance of fighting a group into nil. Before, you could dash around a group if you were appropriately careful, dousing light sources and then jaunting to wait for your critical hit to come up again. This made for an extraordinarily rewarding playstyle wherein your targets had a pretty significant window to GTFO if they wanted to while you get to have the power fantasy of being the wolf darting in and out from the treeline, drawing blood and then retreating until the moment was right.

This change does summarily solve a few glaring issues, and I don't think it'd be the worst thing lost ever considering how nettling it was getting stuck, no matter how few times it's happened to me. Better scouting is also nice, but is still antithetical to the idea of the Nightmare being "restricted" to it's "territory" of doused lights, which was also a very excellent tell when you were stepping into the bad side of town. There was also quite a lot of fun in the mere activity of extending your "territory" by jumping out of jaunt to smash a few lights and spooking folks until The Good Guys showed up to give you a proper fight. 

Jaunt going over light sources without shunting you out does disincentivize this playstyle almost entirely except when setting up ambushes or for the murderboning nightmare in the halls, whom I suppose aren't really affected by this change much at all except for getting owned faster.

I don't have any changes to suggest to fix this except maybe making it a smaller channel? I don't think it's too massive of a buff to be able to go over light sources without being shunted, since realistically all it is is additional scouting and helping the Nightmare escape glitched spots. Maybe keeping the way it is now, but allowing them to shunt to an adjacent tile in full darkness would be better? This solves the glitch spots while preserving the styles I've penned out that play nice with Nightmare's own style.

File: code/modules/spells/spell_types/jaunt/shadow_walk.dm [hunk 1]
```diff
@@ -45,6 +47,18 @@
 		exit_jaunt(cast_on)
 		return
 
+	to_chat(owner, span_warning("You begin entering the shadows..."))
+	animate(owner, alpha = 0, time = shadow_delay)
+	if(!do_after(owner, shadow_delay, owner))
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/modules/spells/spell_types/jaunt/shadow_walk.dm:52]: i couldn't find a good proc for it

**Comment by carlarctg** [comment]
> I will say that this change, while good on paper, would turn any chance of fighting a group into nil. Before, you could dash around a group if you were appropriately careful, dousing light sources and then jaunting to wait for your critical hit to come up again. This made for an extraordinarily rewarding playstyle wherein your targets had a pretty significant window to GTFO if they wanted to while you get to have the power fantasy of being the wolf darting in and out from the treeline, drawing blood and then retreating until the moment was right.
> 
> This change does summarily solve a few glaring issues, and I don't think it'd be the worst thing lost ever considering how nettling it was getting stuck, no matter how few times it's happened to me. Better scouting is also nice, but is still antithetical to the idea of the Nightmare being "restricted" to it's "territory" of doused lights, which was also a very excellent tell when you were stepping into the bad side of town. There was also quite a lot of fun in the mere activity of extending your "territory" by jumping out of jaunt to smash a few lights and spooking folks until The Good Guys showed up to give you a proper fight.
> 
> Jaunt going over light sources without shunting you out does disincentivize this playstyle almost entirely except when setting up ambushes or for the murderboning nightmare in the halls, whom I suppose aren't really affected by this change much at all except for getting owned faster.
> 
> I don't have any changes to suggest to fix this except maybe making it a smaller channel? I don't think it's too massive of a buff to be able to go over light sources without being shunted, since realistically all it is is additional scouting and helping the Nightmare escape glitched spots. Maybe keeping the way it is now, but allowing them to shunt to an adjacent tile in full darkness would be better? This solves the glitch spots while preserving the styles I've penned out that play nice with Nightmare's own style.

I could let you choose at the start between this and the original.

**Comment by HoodMilk** [comment]
The idea of nightmares being a more omnipresent threat, potentially showing up anywhere the lights are out instead of having to make their own territory is interesting. Sort of like a slaughter demon but with darkness instead of blood.

**Comment by carlarctg** [comment]
> The idea of nightmares being a more omnipresent threat, potentially showing up anywhere the lights are out instead of having to make their own territory is interesting. Sort of like a slaughter demon but with darkness instead of blood.

honestly ideally slaughter demons would also have their jaunt inverted because they are just awful rn. but out of scope

**Comment by carlarctg** [comment]
Can the reaction randys speak up or be ignored

**Comment by EnterTheJake** [comment]
It's probably worth a shot, but I won't be surprised if this ends up being more of a nerf than anything.

nightmare relies on hit and run tactics, having a jaunt delay, can potentially cause more trouble.

Why don't just make nightmare be able to jaunt in space?

**Comment by carlarctg** [comment]
theres no way to distinguish space light from normal light afaik. just space tiles from normal tiles which isnt quite enough

**Comment by SkeletalElite** [comment]
Not a huge fan of not being able to quickly jump into and out of jaunt mid fight. You could move the jaunt action button right next to the center so that you can quickly go in and out while fighting and be very hard to hit since you can appear anywhere around your target hit them and disappear, if you get good enough at doing this it basically makes you unfightable once you destroy someones lights but it is pretty difficult and rewarding to do successfully.

With this change the nightmare just becomes slaughter demon 2 a threat that is perpetually untouchable and everytime you manage to damage it even a little bit it just disappears recovers all its health and comes back 1 minute later.

**Comment by nianjiilical** [comment]
this is going to turn nightmare into pre-nerf slaughter demon, an antag that just spends 90% of its time invincible shittalking the crew until it can kill someone who's at half hp in maint

but also im a huge piece of shit who loves slaughter demon so this is a +1 in favor from me

**Comment by MrMelbert** [comment]
I'm torn on this

On one hand it makes them a bit closer to a mini slaughter demon which is maybe more interesting than what they are now. 

On the other hand darkness is a lot lot more common than blood. All it takes is for the light in the captain's office to spawn broken roundstart, and now the nightmare has all the captain's loot, good luck stopping them, lol lmao.

File: code/modules/spells/spell_types/jaunt/shadow_walk.dm [hunk 2]
```diff
@@ -22,21 +24,21 @@
 
 /datum/action/cooldown/spell/jaunt/shadow_walk/enter_jaunt(mob/living/jaunter, turf/loc_override)
 	var/obj/effect/dummy/phased_mob/shadow/shadow = ..()
-	if(istype(shadow))
-		shadow.light_max = light_threshold
 	return shadow
```
Comment (MrMelbert) [inline code/modules/spells/spell_types/jaunt/shadow_walk.dm:27]: This proc is dead just delete it

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by MrMelbert** [comment]
I think they'd be better off if you left jaunt along and gave them the ability to teleport to another section of maintenance.

To me playing Nightmare is fun on maps with well connected maints, but on maps like Delta it sucks because there's no connection between cargo->sec->dorms. Forcing you to rotate the entire station to haunt a new area. 

But if they just had a button which sends them clockwise around maint in order (port -> fore -> starboard -> aft) so you can dictate where to go it'd help that

**Comment by moocowswag** [comment]
This pr is worth trying.

I don't think anyone else has yet to fully grasp the implications of light no longer kicking nightmare out of jaunt. Its not a nerf to counterplay, that counterplay just takes a much more powerful form of pre-emptive knowledge.

nightmare will suddenly have way more information accessible to them and pushes their gameplay loop a bit more towards revenant-like just because they can jaunt and see whats going on in lit areas without risking themselves.

If you get caught by a group of people and cant jaunt out its your fault for not scouting the area pre-emptively, there should always be at least 1 airlock between you and someone that can hurt you which is more than enough time to jaunt away.


A better argument against the pr is not that it will be weaker (because it won't be in the right hands) but if we want revenant 2.0.

**Comment by MrMelbert** [comment]
Dunno, I guess you'd give them a radial, or embed it into their antag panel

And then for those without the antag datum they'd just get one or the other randomly

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm fine with an additional power that lets them rotate between maints with chargeup/cooldown if the issue is disconnected maintenance I guess, but nightmares arent SUPPOSE to be main characters. You are SUPPOSE to lose if you are checked by a competent group of people because your role is a conflict that forces them to take attention off the actual antags.

It is called a side antag for a reason, Id understand if the issue is youre losing 1 on 1s but if you lose to a group of people that is purposeful and should not be seen as a flaw to fix for the role.

**Pull request closed without merging**